jupyterhub/tests/selenium/test_browser.py

```diff
@@ -209,36 +225,686 @@ async def test_open_url_login(
+                )
+                for i in range(len(logs)):
+                    progress_log = logs[i].text
+                    logs_list.append(progress_log)
+                if progress_messages == "":
+                    assert percent == "0%"
+                    assert int(len(logs_list)) == 0
+                if progress_messages == "Server requested":
+                    assert percent == "0%"
+                    assert int(len(logs_list)) == 1
+                    assert str(logs_list[0]) == "Server requested"
+                if progress_messages == "Spawning server...":
+                    assert percent == "50%"
+                    assert int(len(logs_list)) == 2
+                    assert str(logs_list[0]) == "Server requested"
+                    assert str(logs_list[1]) == "Spawning server..."
+                if "Server ready at" in progress_messages:
+                    assert (
+                        f"Server ready at {app.base_url}user/{user}/"
+                        in progress_messages
+                    )
+                    assert percent == "100%"
+                    assert int(len(logs_list)) == 3
+                    assert str(logs_list[0]) == "Server requested"
+                    assert str(logs_list[1]) == "Spawning server..."
+                    assert (
+                        str(logs_list[2])
+                        == f"Server ready at {app.base_url}user/{user}/"
+                    )
+                    assert str(logs_list[2]) == progress_messages
+        except NoSuchElementException:
+            break
+        except StaleElementReferenceException:
+            break
+        await asyncio.sleep(0.001)
+
+
+async def test_spawn_pending_launch_server(
+    app, browser, user="test_user1", pass_w="test_user1"
+):
+    """verify that after a successful launch server via the spawn-pending page
+    the user should see two buttons on the home page"""
+
+    await open_spawn_pending(app, browser, user, pass_w)
+    button = browser.find_element(*SpawningPageLocators.BUTTON_START_SERVER)
+    u = app.users[orm.User.find(app.db, user)]
+    click(browser, SpawningPageLocators.BUTTON_START_SERVER)
+    await webdriver_wait(browser, EC.staleness_of(button))
+    # checking that server is running and two butons present on the home page
+    home_page = url_path_join(public_host(app), ujoin(app.base_url, "hub/home"))
+    await in_thread(browser.get, home_page)
+    assert u.spawner.ready
+    assert is_displayed(browser, HomePageLocators.BUTTON_START_SERVER)
+    assert is_displayed(browser, HomePageLocators.BUTTON_STOP_SERVER)
+
+
+# HOME PAGE
+
+
+async def open_home_page(app, browser, user, pass_w):
+    """function to open the home page"""
+
+    home_page = url_escape(app.base_url) + "hub/home"
+    await open_url(app, browser, param="/login?next=" + home_page)
+    await login(browser, user, pass_w)
+    while '/hub/home' not in browser.current_url:
+        await asyncio.sleep(0.5)
```

Comment:
This can be shorter:

```suggestion
        await asyncio.sleep(0.1)
```